src/services/transaction-apis.ts

```diff
@@ -18,15 +20,25 @@ export function buildTransactionServiceUrl ({
   explorerAPI,
   transactionIdPlaceholder = TRANSACTION_ID_PLACEHOLDER,
   transactionId = '',
-  isTestApi = false
+  chain = null
 }: {
   explorerAPI: ExplorerAPI;
   transactionIdPlaceholder?: string;
   transactionId?: string;
-  isTestApi?: boolean;
+  chain?: SupportedChains;
 }): string {
-  const { serviceURL } = explorerAPI;
-  let apiUrl = typeof serviceURL === 'string' ? serviceURL : (isTestApi ? serviceURL.test : serviceURL.main);
+  const { serviceURL, getTransactionServiceURL } = explorerAPI;
+  let apiUrl: string;
+  if (typeof serviceURL === 'string') {
+    apiUrl = serviceURL;
+  } else if (typeof serviceURL === 'object') {
+    const isTestApi = chain ? isTestChain(chain) : false;
+    apiUrl = isTestApi ? serviceURL.test : serviceURL.main;
+  } else if (typeof getTransactionServiceURL === 'function') {
+    apiUrl = getTransactionServiceURL(chain);
```

Comment:
I think this is getting too complicated. I will take another look later to see how we can simplify the surface.